realm/realm-library/src/main/java/io/realm/ObjectChangeSet.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+/**
+ * Information about the changes made to an object.
+ *
+ * @see RealmObject#addChangeListener(RealmObjectChangeListener) .
+ */
+public interface ObjectChangeSet {
+
+    /**
+     * @return true if the object has been deleted from the Realm.
+     */
+    boolean isDeleted();
+
+    /**
+     * @return the names of changed fields if the object still exists and there are field changes. Returns {@code null}
+     * if the object has been deleted.
+     */
+    String[] getChangedFields();
```

Comment:
I would think a very common pattern is:

```
if (changeset.isFieldChanged("name")) {
  updateName(changeSet.getChange(String.class, "name").newValue());
}
```

That should be as easy as possible.

Agree, that it gets slightly ambigious with deleted objects, but `getChangedFields()` would also return nothing, so I think that will be acceptable?